src/components/PostDetails.tsx

```diff
@@ -158,7 +293,7 @@
         )}
 
         {loaded && !hasError && visible && (
-          <NewCommentForm onSubmit={addComment} />
+          <NewCommentForm onSubmit={handleAddComment} />
```

Comment:
The `NewCommentForm` is displayed based on the `visible` state. Ensure that this state is correctly managed and reset when necessary, especially when switching between different posts.